crates/wasmparser/src/readers/core/imports.rs

```diff
@@ -38,6 +38,12 @@ pub enum TypeRef {
     ///
     /// The value is an index in the types index space.
     Tag(TagType),
+    /// The type is a module.
+    ///
+    /// The value is an index into the the type section.
+    ///
+    /// This variant is currently only used with the component model proposal.
+    Module(u32),
```

Comment:
Ah in my comment about `read_type` above where I mentioned I don't think we should change the AST of core wasm this is another location which I think would be better to avoid if we can.

Reply:
These component-model extensions to this core enumeration have been removed so as to not affect core parsing.

We now have additional enumerations for the component aliases to account for this; it also helped me identify that I was missing an implementation of encoding/parsing/validating outer aliases to core types.